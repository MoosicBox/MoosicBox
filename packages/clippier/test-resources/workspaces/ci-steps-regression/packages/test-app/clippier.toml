# Test configuration for ci-steps regression
# This reproduces the bug where ci-steps with both command and toolchain
# would not properly export ciToolchains in the matrix output

[[ci-steps]]
command   = "mkdir packages/app/tauri/dist"
toolchain = "free_disk_space"

[[ci-steps]]
command   = "pnpm install"
features  = ["frontend"]
toolchain = "pnpm"

[[ci-steps]]
features  = ["frontend", "tauri"]
toolchain = "node"

[[ci-steps]]
command = "cargo build"

[[config]]
os = "ubuntu"

[[config.dependencies]]
command = "apt-get update && apt-get install -y build-essential"
