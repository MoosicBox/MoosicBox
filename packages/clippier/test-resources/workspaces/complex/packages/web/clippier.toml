[parallelization]
chunked = 2

[env]
FRONTEND_BUILD = { value = "production", features = ["frontend"] }
WEB_PORT       = "8080"

[[config]]
name = "frontend-build"
os   = "ubuntu"

[[config.dependencies]]
command  = "apt-get update && apt-get install -y nodejs npm"
features = ["frontend"]

[[config.ci-steps]]
command  = "npm install"
features = ["frontend"]

[[config.ci-steps]]
command  = "npm run build"
features = ["frontend"]

[[config]]
name = "ssr-build"
os   = "ubuntu"

[[config.dependencies]]
command  = "apt-get update && apt-get install -y nodejs npm"
features = ["ssr"]

[config.env]
SSR_ENABLED = "true"
