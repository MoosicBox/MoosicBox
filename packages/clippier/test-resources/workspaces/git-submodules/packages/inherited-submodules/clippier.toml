git-submodules = true

[[config]]
os = "ubuntu"

[[config]]
os = "macos"
