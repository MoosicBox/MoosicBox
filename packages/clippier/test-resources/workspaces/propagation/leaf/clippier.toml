git-submodules = true

[env]
BAR_VERSION = "2.0"
FOO_PATH    = "/usr/local/foo"

[[config]]
os = "ubuntu"

[[config.dependencies]]
command = "apt-get install -y libfoo-dev"

[[config.dependencies]]
command = "apt-get install -y libbar-dev"

[[config.ci-steps]]
command = "curl -o vectors.tar.gz https://example.com/vectors.tar.gz"

[[config.ci-steps]]
command = "tar -xzf vectors.tar.gz"
