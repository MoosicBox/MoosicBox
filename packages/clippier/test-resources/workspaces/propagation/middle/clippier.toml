[env]
MIDDLE_VAR = "middle_value"
BAR_VERSION = "3.0"

[[config]]
os = "ubuntu"

[[config.dependencies]]
command = "apt-get install -y libmiddle-dev"

[[config.ci-steps]]
command = "echo 'Setting up middle'"
