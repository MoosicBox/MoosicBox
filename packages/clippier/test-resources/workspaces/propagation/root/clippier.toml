[env]
FOO_PATH = "/override/foo"
ROOT_VAR = "root_value"

[[config]]
os = "ubuntu"

[[config.ci-steps]]
command = "cargo test"
