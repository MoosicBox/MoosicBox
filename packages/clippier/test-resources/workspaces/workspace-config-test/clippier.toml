# Workspace-level defaults
nightly = false

[env]
RUST_BACKTRACE = "1"
WORKSPACE_VAR  = "from_workspace"

[[ci-steps]]
command = "echo workspace ci step"

[[dependencies]]
command = "apt-get install workspace-dep"
