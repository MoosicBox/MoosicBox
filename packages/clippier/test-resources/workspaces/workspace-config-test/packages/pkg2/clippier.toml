# Package-level overrides
[env]
PACKAGE_VAR = "from_package"

[[config]]
os = "ubuntu"

[[config.dependencies]]
command = "apt-get install package-dep"
