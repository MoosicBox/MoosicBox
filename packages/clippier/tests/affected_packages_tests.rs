use clippier::AffectedPackageInfo;
use clippier_test_utilities::test_resources::load_test_workspace;
use std::fs;
use tempfile::TempDir;

/// Create a test workspace for affected packages testing
fn create_affected_packages_workspace() -> TempDir {
    let temp_dir = TempDir::new().expect("Failed to create temp directory");

    // Create workspace Cargo.toml
    let workspace_toml = r#"
[workspace]
members = [
    "packages/core",
    "packages/models",
    "packages/api",
    "packages/web",
    "packages/cli",
    "packages/shared-utils"
]
"#;
    fs::write(temp_dir.path().join("Cargo.toml"), workspace_toml)
        .expect("Failed to write workspace Cargo.toml");

    // Core package - foundation
    let core_dir = temp_dir.path().join("packages/core");
    fs::create_dir_all(core_dir.join("src")).expect("Failed to create core directory");
    let core_cargo = r#"
[package]
name = "core"
version = "0.1.0"
edition = "2021"

[dependencies]
shared-utils = { workspace = true }
"#;
    fs::write(core_dir.join("Cargo.toml"), core_cargo).expect("Failed to write core Cargo.toml");
    fs::write(core_dir.join("src/lib.rs"), "// core").expect("Failed to write core lib.rs");

    // Models package - depends on core
    let models_dir = temp_dir.path().join("packages/models");
    fs::create_dir_all(models_dir.join("src")).expect("Failed to create models directory");
    let models_cargo = r#"
[package]
name = "models"
version = "0.1.0"
edition = "2021"

[dependencies]
core = { workspace = true }
"#;
    fs::write(models_dir.join("Cargo.toml"), models_cargo)
        .expect("Failed to write models Cargo.toml");
    fs::write(models_dir.join("src/lib.rs"), "// models").expect("Failed to write models lib.rs");

    // API package - depends on models
    let api_dir = temp_dir.path().join("packages/api");
    fs::create_dir_all(api_dir.join("src")).expect("Failed to create api directory");
    let api_cargo = r#"
[package]
name = "api"
version = "0.1.0"
edition = "2021"

[dependencies]
models = { workspace = true }
core = { workspace = true }
"#;
    fs::write(api_dir.join("Cargo.toml"), api_cargo).expect("Failed to write api Cargo.toml");
    fs::write(api_dir.join("src/lib.rs"), "// api").expect("Failed to write api lib.rs");

    // Web package - depends on api
    let web_dir = temp_dir.path().join("packages/web");
    fs::create_dir_all(web_dir.join("src")).expect("Failed to create web directory");
    let web_cargo = r#"
[package]
name = "web"
version = "0.1.0"
edition = "2021"

[dependencies]
api = { workspace = true }
"#;
    fs::write(web_dir.join("Cargo.toml"), web_cargo).expect("Failed to write web Cargo.toml");
    fs::write(web_dir.join("src/lib.rs"), "// web").expect("Failed to write web lib.rs");

    // CLI package - depends on api and models
    let cli_dir = temp_dir.path().join("packages/cli");
    fs::create_dir_all(cli_dir.join("src")).expect("Failed to create cli directory");
    let cli_cargo = r#"
[package]
name = "cli"
version = "0.1.0"
edition = "2021"

[dependencies]
api = { workspace = true }
models = { workspace = true }
"#;
    fs::write(cli_dir.join("Cargo.toml"), cli_cargo).expect("Failed to write cli Cargo.toml");
    fs::write(cli_dir.join("src/main.rs"), "fn main() {}").expect("Failed to write cli main.rs");

    // Shared utils package - standalone
    let utils_dir = temp_dir.path().join("packages/shared-utils");
    fs::create_dir_all(utils_dir.join("src")).expect("Failed to create utils directory");
    let utils_cargo = r#"
[package]
name = "shared-utils"
version = "0.1.0"
edition = "2021"
"#;
    fs::write(utils_dir.join("Cargo.toml"), utils_cargo).expect("Failed to write utils Cargo.toml");
    fs::write(utils_dir.join("src/lib.rs"), "// shared utils")
        .expect("Failed to write utils lib.rs");

    temp_dir
}

#[test]
fn test_find_affected_packages_direct_change() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/core/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["core"]);
}

#[test]
fn test_find_affected_packages_leaf_change() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/web/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["web"]);
}

#[test]
fn test_find_affected_packages_multiple_files() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec![
        "packages/core/src/lib.rs".to_string(),
        "packages/shared-utils/src/lib.rs".to_string(),
    ];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["core", "shared-utils"]);
}

#[test]
fn test_find_affected_packages_with_reasoning() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/core/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages_with_reasoning(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(
        packages,
        vec![AffectedPackageInfo {
            name: "core".to_string(),
            reasoning: Some(vec![
                "Contains changed file: packages/core/src/lib.rs".to_string()
            ])
        }]
    );
}

#[test]
fn test_find_affected_packages_cargo_toml_change() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/core/Cargo.toml".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["core"]);
}

#[test]
fn test_find_affected_packages_nested_path() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/api/src/handlers/mod.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["api"]);
}

#[test]
fn test_affected_packages_complex_dependency_chain() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/core/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["core"]);
}

#[test]
fn test_find_affected_packages_mixed_changes() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec![
        "packages/core/src/lib.rs".to_string(),
        "packages/api/Cargo.toml".to_string(),
        "packages/web/src/components/mod.rs".to_string(),
    ];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let packages = result.unwrap();
    assert_eq!(packages, vec!["api", "core", "web"]);
}

#[test]
fn test_single_package_affected_check() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/api/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());

    let all_affected = result.unwrap();
    assert_eq!(all_affected, vec!["api"]);
}

#[test]
fn test_find_affected_packages_no_changes() {
    let temp_dir = create_affected_packages_workspace();

    let changed_files = vec!["README.md".to_string(), "docs/guide.md".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files)
        .expect("Failed to find affected packages");

    // No packages should be affected by non-package files
    assert_eq!(result.len(), 0);
}

#[test]
fn test_find_affected_packages_workspace_root_change() {
    let temp_dir = create_affected_packages_workspace();

    let changed_files = vec!["Cargo.toml".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files)
        .expect("Failed to find affected packages");

    // Workspace root changes typically don't affect individual packages
    // unless the packages are under the workspace root directly
    assert_eq!(result.len(), 0);
}

#[test]
fn test_find_affected_packages_partial_path_match() {
    let temp_dir = create_affected_packages_workspace();

    // Create a file that partially matches a package name but is not in the package
    let false_dir = temp_dir.path().join("packages-backup");
    fs::create_dir_all(&false_dir).expect("Failed to create false directory");
    fs::write(false_dir.join("core-backup.rs"), "// backup").expect("Failed to write false file");

    let changed_files = vec!["packages-backup/core-backup.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files)
        .expect("Failed to find affected packages");

    // Should not affect any packages since the file is not in a package directory
    assert_eq!(result.len(), 0);
}

#[test]
fn test_find_affected_packages_case_sensitivity() {
    let temp_dir = create_affected_packages_workspace();

    let changed_files = vec!["PACKAGES/core/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files)
        .expect("Failed to find affected packages");

    // Should not match due to case sensitivity (PACKAGES vs packages)
    assert_eq!(result.len(), 0);
}

#[cfg(feature = "git-diff")]
#[test]
fn test_find_affected_packages_with_external_deps() {
    use clippier_test_utilities::test_resources::load_cargo_lock_for_git_diff;

    let (temp_dir, _) = load_test_workspace("complex");

    // Test that external dependency analysis utilities are available
    // This tests the git-diff feature integration
    let _cargo_lock = load_cargo_lock_for_git_diff("basic", "simple");

    // For now, just test that basic affected packages functionality works
    let changed_files = vec!["packages/api/src/lib.rs".to_string()];
    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);

    assert!(result.is_ok());
    let packages = result.unwrap();

    // Since this is a stub implementation, we expect empty results
    assert_eq!(packages, vec!["api"]);
}

#[test]
fn test_empty_changed_files() {
    let temp_dir = create_affected_packages_workspace();

    let changed_files: Vec<String> = vec![];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files)
        .expect("Failed to find affected packages");

    // No files changed, no packages affected
    assert_eq!(result.len(), 0);
}

#[test]
fn test_direct_file_changes() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/api/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_transitive_impact_analysis() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/core/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_multiple_file_changes() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec![
        "packages/core/src/lib.rs".to_string(),
        "packages/shared-utils/src/lib.rs".to_string(),
    ];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_complex_dependency_chains() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/api/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_affected_with_reasoning() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/models/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages_with_reasoning(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_nested_path_edge_cases() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec![
        "packages/api/src/handlers/mod.rs".to_string(),
        "packages/api/tests/integration.rs".to_string(),
    ];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_partial_path_matches() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/ap/file.rs".to_string()]; // Partial match

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_case_sensitivity() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["PACKAGES/API/src/lib.rs".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_cargo_toml_vs_source_changes() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["packages/api/Cargo.toml".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_empty_change_sets() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files: Vec<String> = vec![];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

#[test]
fn test_workspace_root_changes() {
    let (temp_dir, _) = load_test_workspace("complex");
    let changed_files = vec!["Cargo.toml".to_string(), "Cargo.lock".to_string()];

    let result = clippier::find_affected_packages(temp_dir.path(), &changed_files);
    assert!(result.is_ok());
}

// Snapshot tests with proper JSON serialization
#[test]
fn test_direct_file_changes_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": ["packages/api/src/lib.rs"],
        "directly_affected": ["api"],
        "reasoning": {
            "api": ["Contains changed file: packages/api/src/lib.rs"]
        }
    });

    insta::assert_yaml_snapshot!("direct_file_changes", test_data);
}

#[test]
fn test_transitive_impact_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": ["packages/core/src/lib.rs"],
        "directly_affected": ["core"],
        "transitively_affected": ["models", "api", "web", "cli"],
        "dependency_chain": {
            "models": "depends on core",
            "api": "depends on models",
            "web": "depends on api",
            "cli": "depends on api"
        }
    });

    insta::assert_yaml_snapshot!("transitive_impact", test_data);
}

#[test]
fn test_multiple_file_changes_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": [
            "packages/core/src/lib.rs",
            "packages/shared-utils/src/lib.rs"
        ],
        "directly_affected": ["core", "shared-utils"],
        "note": "Multiple files can affect different dependency trees"
    });

    insta::assert_yaml_snapshot!("multiple_file_changes", test_data);
}

#[test]
fn test_complex_dependency_chains_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "dependency_graph": {
            "core": [],
            "shared-utils": [],
            "models": ["core"],
            "api": ["models", "shared-utils"],
            "web": ["api"],
            "cli": ["api"]
        },
        "changed_api_affects": ["web", "cli"]
    });

    insta::assert_yaml_snapshot!("complex_dependency_chains", test_data);
}

#[test]
fn test_affected_reasoning_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": ["packages/models/src/lib.rs"],
        "reasoning_chain": {
            "models": ["Contains changed file: packages/models/src/lib.rs"],
            "api": ["Depends on affected package: models"],
            "web": ["Depends on affected package: api"],
            "cli": ["Depends on affected package: api"]
        },
        "note": "Each package includes reasoning for why it's affected"
    });

    insta::assert_yaml_snapshot!("affected_reasoning", test_data);
}

#[test]
fn test_nested_path_edge_cases_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": [
            "packages/api/src/handlers/mod.rs",
            "packages/api/tests/integration.rs",
            "packages/api/benches/benchmark.rs"
        ],
        "affected_packages": ["api"],
        "note": "All files within package directory should be detected"
    });

    insta::assert_yaml_snapshot!("nested_path_edge_cases", test_data);
}

#[test]
fn test_partial_path_matches_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    // Test edge case: partial path matches should not affect packages
    // assert!(true); // This is testing that "packages/ap" doesn't match "packages/api"
}

#[test]
fn test_case_sensitivity_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    // Test case sensitivity behavior
    // assert!(true); // This depends on filesystem case sensitivity
}

#[test]
fn test_empty_change_sets_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": [],
        "affected_packages": [],
        "note": "No changes should result in no affected packages"
    });

    insta::assert_yaml_snapshot!("empty_change_sets", test_data);
}

#[test]
fn test_workspace_root_changes_snapshot() {
    let (_temp_dir, _) = load_test_workspace("complex");

    let test_data = serde_json::json!({
        "changed_files": ["Cargo.toml", "Cargo.lock"],
        "affected_packages": [],
        "note": "Workspace-level changes don't map to specific packages"
    });

    insta::assert_yaml_snapshot!("workspace_root_changes", test_data);
}

#[cfg(feature = "git-diff")]
#[test]
fn test_external_dependency_integration() {
    let (_temp_dir, _) = load_test_workspace("complex");

    // Test integration with external dependency analysis
    let test_data = serde_json::json!({
        "external_deps_changed": ["serde", "tokio"],
        "packages_using_serde": ["core", "models", "shared-utils"],
        "packages_using_tokio": ["core", "api"],
        "transitively_affected": ["api", "web", "cli"],
        "reasoning": {
            "core": ["External dependency changes: serde, tokio"],
            "models": ["Depends on affected package: core"],
            "api": ["External dependency changes: tokio", "Depends on affected package: models"]
        }
    });

    insta::assert_yaml_snapshot!("external_deps_integration", test_data);
}
