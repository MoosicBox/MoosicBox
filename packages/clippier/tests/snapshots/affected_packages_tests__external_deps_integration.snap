---
source: packages/clippier/tests/affected_packages_tests.rs
expression: test_data
---
external_deps_changed:
  - serde
  - tokio
packages_using_serde:
  - core
  - models
  - shared-utils
packages_using_tokio:
  - core
  - api
reasoning:
  api:
    - "External dependency changes: tokio"
    - "Depends on affected package: models"
  core:
    - "External dependency changes: serde, tokio"
  models:
    - "Depends on affected package: core"
transitively_affected:
  - api
  - web
  - cli
